Tasks/Common/azure-arm-rest/webClient.ts

```diff
@@ -0,0 +1,86 @@
+import tl = require('vsts-task-lib/task');
+import util = require("util")
+var httpClient = require('vso-node-api/HttpClient');
+var httpCallbackClient = new httpClient.HttpCallbackClient(tl.getVariable("AZURE_HTTP_USER_AGENT"));
+
+export class WebRequest {
+    public method: string;
+    public uri: string;
+    public body: any;
+    public headers: any;
+}
+
+export class WebResponse {
+    public statusCode: number;
+    public statusMessage: string;
+    public headers: any;
+    public body: any;
+}
+
+export class WebRequestOptions {
+    public retriableErrorCodes: string[];
+    public retryCount: number;
+    public retryIntervalInSeconds: number;
+}
+
+export async function sendRequest(request: WebRequest, options?: WebRequestOptions): Promise<WebResponse> {
+    let i = 0;
+    let retryCount = options && options.retryCount ? options.retryCount : 5;
+    let retryIntervalInSeconds = options && options.retryIntervalInSeconds ? options.retryIntervalInSeconds: 5;
+    let retriableErrorCodes = options && options.retriableErrorCodes ? options.retriableErrorCodes : ["ETIMEDOUT"];
```

Comment:
is there an error code for 502? that is something we can also think of for retry